Sources/Sentry/SentryScope.m

```diff
@@ -319,6 +326,60 @@ - (void)addAttachment:(SentryAttachment *)attachment
     }
 }
 
+- (void)setTransaction:(id<SentrySpan>)transaction forKey:(NSString *)key
```

Comment:
is the idea that all of those methods are public APIs or only of internal usage?
because for auto instrumentation, either transaction's name or as you call, scopeKey, the naming could be dynamically created and I don't see how they would use those methods.

Reply:
Yes, the idea is that these methods are public APIs. Internally we could use the transaction name as key we are just given the user a few extra options to use.

We have some extra problem here. In cocoa all transactions are spans, not all spans have name. The user could use a child span in the setTransaction method.

Reply:
I'd not bother with this now, we could simply write to the docs that only the root span should be bound to the Scope, or, adding an overload like Java, that allows or not binding the transaction to the scope automatically (when creating a transaction instead of calling configureScope), so they don't have to do it manually.

Reply:
cc @bruno-garcia @philipphofmann

Reply:
I would also make this method internal. Then we are sure that the transaction has a name.